authorizer.go

```diff
@@ -46,19 +46,38 @@ type authorizer struct {
 
 var _ Authorizer = (*authorizer)(nil)
 
-func NewVerifier(b *Biscuit) (Authorizer, error) {
-	baseWorld := datalog.NewWorld()
+type AuthorizerOption func(w *authorizer)
 
-	return &authorizer{
+func WithBaseWorld(baseWorld *datalog.World) AuthorizerOption {
+	return func(a *authorizer) {
+		a.baseWorld = baseWorld
+	}
+}
+
+func WithBaseSymbolTable(syms *datalog.SymbolTable) AuthorizerOption {
+	return func(a *authorizer) {
+		a.baseSymbols = syms.Clone()
+	}
+}
+
+func NewVerifier(b *Biscuit, opts ...AuthorizerOption) (Authorizer, error) {
+	a := &authorizer{
 		biscuit:      b,
-		baseWorld:    baseWorld,
-		world:        baseWorld.Clone(),
-		symbols:      defaultSymbolTable.Clone(),
+		baseWorld:    datalog.NewWorld(),
 		baseSymbols:  defaultSymbolTable.Clone(),
 		checks:       []Check{},
 		policies:     []Policy{},
 		block_worlds: []*datalog.World{},
-	}, nil
+	}
+
+	for _, opt := range opts {
+		opt(a)
```

Comment:
is that a common pattern in Go to makes structs configurable? It feels a bit weird to pass functions that modify inner fields. While it would make sense to move towards an Authorizer implementation that hides inner details, because they can break easily

Reply:
> is that a common pattern in Go to makes structs configurable?

Yes, it seems to be. It's an alternative to passing a config struct or exposing certain fields, and is used a few other places in this repo (e.g. [datalog.go](https://github.com/biscuit-auth/biscuit-go/blob/main/datalog/datalog.go#L327) and [builder.go](https://github.com/biscuit-auth/biscuit-go/blob/main/builder.go#L55)).